docs/zeebe/dynamic_scaling.md

```diff
@@ -0,0 +1,209 @@
+- **Decision Definitions**: DMN decision tables and DRD definitions  
+- **Form Definitions**: forms linked to user tasks
+- **Global Configuration**: Routing state and partition configuration
+
+**Important**: Partition-specific state (active process instances, jobs, message subscriptions) remains on the original partitions and is **not** redistributed.
+
+A sequence diagram of the redistribution process can be found below:
+
+```mermaid
+sequenceDiagram
+autonumber
+participant CoordinatorNode
+participant P-1
+participant P-1-Snapshot
+participant P-ith
+participant P-NEW
+
+CoordinatorNode ->>+ CoordinatorNode: Add ClusterChangeOperations to ClusterPlan
+
+CoordinatorNode ->> P-1: Send SCALE_UP (w/ desiredPartitionCount)
+P-1 ->> P-1: [SCALING_UP] Add desiredPartitionCount to RoutingInfo: <br> new partitions are included in distribution (only enqueued)
+
+CoordinatorNode ->>+ P-NEW: Bootstrap new partition from snapshot of P1
+P-NEW ->> P-1: Get snapshot for bootstrap
+P-1 ->> P-1: Take a snapshot such that processedPosition > position of SCALING_UP event
+
+CoordinatorNode ->> CoordinatorNode: AwaitRedistributionCompletion
+
+CoordinatorNode ->> P-1: Redistribution completed?
+P-1 ->> CoordinatorNode: No
+
+P-1 ->> P-NEW: Snapshot for bootstrap with SCALING_UP event
+P-NEW ->>- P-1: Partition NEW Bootstrapped
+P-1 ->> P-1: Start distributing pending commands
+
+loop for command in enqueued_distribution_commands
+  P-1 ->> P-NEW: send command 
+end
+
+P-1 ->> P-2: Partition i Bootstrapped
+CoordinatorNode ->> P-1: Redistribution completed?
+P-1 ->> CoordinatorNode: Yes
+CoordinatorNode ->>- CoordinatorNode : Update RoutingState.RequestHandling
+
+P-1 ->> P-ith: Distribution event at b_e + n
+P-1 ->> P-NEW: Distribution event at b_e + n
+```
+
+### Command Distribution Mechanism
+
+The redistribution phase uses [Zeebe's command distribution](./generalized_distribution.md) system to ensure reliable deployment of resources across partitions:
+
+1. **Command Enqueueing**: When scaling begins, deployment commands for new partitions are enqueued but not immediately sent
+2. **Bootstrap Completion Signal**: Each new partition signals completion of bootstrap to partition 1
+3. **Distributed Deployment**: When a partition is bootstrapped, partition 1 distributes queued deployment commands to the partitions (including existing ones)
+4. **Acknowledgment**: Receiving partitions acknowledge successful processing of deployment commands
+5. **Retry Logic**: Failed distributions are automatically retried with exponential backoff
+
+### Scaling steps
+
+1. A client sends a [ClusterConfigurationManagementRequest](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/ClusterConfigurationManagementApi.java) as a HTTP PATCH request at `/actuator/cluster/` on the management port.
+
+   If new brokers need to be added, it can be done in the same command, by specifying the ids of the brokers to add:
```

Comment:
:question:  Does the new partitions goes to new brokers or we always re-distribute partitions using round-robin? Would be good to also document this.